src/type-factory/union.type-factory.ts

```diff
@@ -0,0 +1,20 @@
+import * as graphql from 'graphql';
+import { MetadataStorage } from '../metadata-storage';
+import { UnionTypeMetadata } from '../metadata';
+import { objectTypeFactory } from '../object_type_factory';
+
+export function unionTypeFactory(name: string, isInput: boolean): graphql.GraphQLUnionType {
+  return MetadataStorage.getUnionMetadata()
+    .filter(union => union.name === name)
+    .map(union => {
+      return new graphql.GraphQLUnionType({
+        description: union.description,
+        name: union.name,
+        resolveType: union.resolver,
+        types: union.types
+          .map(type => objectTypeFactory(type, isInput))
+          .filter(_ => _), //filter null values
+      });
+    })
+    .find((_, index) => index === 0) || null;
```

Comment:
What is the purpose of this `|| null`?

Reply:
just to be explicit in the return type instead of returning `undefined`. I actually could have set the function return type to be `graphql.GraphQLUnionType | undefined ` instead 👍